### ISSUE TITLE
open: permission denied

### ISSUE THREAD

**Issue by shyiko** [shyiko/jabba#190]
https://github.com/travis-ci/travis-ci/issues/9572#issuecomment-386788932

**Comment by shyiko** [comment]
It appears there is more to the fix than simply https://github.com/shyiko/jabba/commit/393dc18a6ee2585eecc67200bc10f6d3b9c8d8c3 (not reproducable locally).

**Comment by shyiko** [comment]
(hopefully) Fixed in [0.9.6](https://github.com/shyiko/jabba/blob/master/CHANGELOG.md#096---2018-05-05). 

Verified using the same Travis CI configuration as in [eed3si9n/sbt (wip/discover-java-home)](https://github.com/eed3si9n/sbt/blob/d1a61303bf08f14aae6bb8e393efaaf369769247/.travis.yml) (build succeeded 10 times out of 10).

@eed3si9n please let me know if you experience any problems. :bowing_man:

**Comment by shyiko** [comment]
https://travis-ci.org/eed3si9n/sbt/jobs/375357773#L448 failed

**Comment by shyiko** [comment]
@eed3si9n sorry for all the back and forth but do you think you could try
- jabba 0.10.0 (goroutine performing the installation is now locked to a single OS thread. Normally, this should not be required, but as I'm unable to reproduce the problem myself (tried both locally and on Travis CI (including the same docker image)) I can only guess at this point)
- switching from "Container on EC2" to "Virtual machine on GCE" (https://docs.travis-ci.com/user/reference/overview/#Virtualisation-Environment-vs-Operating-System)

**Comment by eed3si9n** [comment]
The previous failures may be been related to my Travis CI cache.

I saw your message above, and added a new commit on sbt 0.10.0, and it failed on the first try:

```
1.94s$ curl -sL https://raw.githubusercontent.com/shyiko/jabba/0.10.0/install.sh | bash && . ~/.jabba/jabba.sh
Installing v0.10.0...
Adding source string to /home/travis/.bashrc
Adding source string to /home/travis/.bash_profile
Installation completed
(if you have any problems please report them at https://github.com/shyiko/jabba/issues)
13.10s$ /home/travis/.jabba/bin/jabba install openjdk@1.10
Downloading openjdk@1.10.0 (https://download.java.net/java/GA/jdk10/10.0.1/fb4372174a714e6b8c52526dc134031e/10/openjdk-10.0.1_linux-x64_bin.tar.gz) 
204885801/204885801
Extracting /tmp/jabba-d-109463111 to /home/travis/.jabba/jdk/openjdk@1.10.0 
open /home/travis/.jabba/jdk/openjdk@1.10.0/legal/java.base/ADDITIONAL_LICENSE_INFO: permission denied 
```

But it came to me that I might be causing this problem by caching

```yaml
cache:
  directories:
    - $HOME/.ivy2/cache
    - $HOME/.sbt/boot
    - $HOME/.jabba
```

so I wiped out Travis CI cache, and it seemed to have worked:

```
14.16s$ /home/travis/.jabba/bin/jabba install openjdk@1.10
Downloading openjdk@1.10.0 (https://download.java.net/java/GA/jdk10/10.0.1/fb4372174a714e6b8c52526dc134031e/10/openjdk-10.0.1_linux-x64_bin.tar.gz) 
204885801/204885801
Extracting /tmp/jabba-d-157287413 to /home/travis/.jabba/jdk/openjdk@1.10.0 
openjdk@1.10 -> /home/travis/.jabba/jdk/openjdk@1.10.0 
```

**Comment by shyiko** [comment]
Right... cache must have contained files without write permission (e.g. 0444) from jabba pre-0.9.5 (jabba 0.9.5+ makes sure write flag is set, but if the file (w/o write bit) already exists `open(2)` fails). This (along with [incorrect install-check fixed in 0.10.1](https://github.com/shyiko/jabba/blob/master/CHANGELOG.md#0101---2018-05-07)) explains everything. 

Thank you for helping gunning these bugs down!
:bowing_man:

**Comment by eed3si9n** [comment]
Nice. Thanks for following up on all the loose ends.
